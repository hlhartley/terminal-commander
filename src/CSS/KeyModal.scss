@import '_variables.scss';

.key-modal {
    // animation: popUp .5s ease 0s 1 normal;     
    background-color: #ffc8c8;
    border: 3px solid black;
    bottom: 0;
    height: 25%;
    left: 0;
    margin: auto;
    overflow: scroll;
    padding: 35px;
    position: absolute;
    right: 0;
    top: -225px;
    width: 65%;
    border-radius: 15px;
    z-index: 10;
    display: none;
}