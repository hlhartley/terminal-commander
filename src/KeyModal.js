import React from 'react';
import './CSS/main.scss';

const KeyModal = (props) => {
    return (
     <div className='key-modal'>
      <div className='button-container'>
        <button onClick={props.toggleModal}>x</button>
         </div>
         <h2 className='modal-title'><i class="fas fa-key"></i>Key:</h2>
            <ul className='modal-list fa-ul'>
                <li><i class="fa-li fas fa-arrow-right"></i><span className='skyblue'>[file]</span> = file name</li>
                <li><i class="fa-li fas fa-arrow-right"></i><span className='skyblue'>[directory]</span> = directory name</li>
                <li><i class="fa-li fas fa-arrow-right"></i><span className='skyblue'>[file extension]</span> = file extension type (ex. .html, .css, .txt)</li>
            </ul>
     </div>
    );
  }

export default KeyModal;