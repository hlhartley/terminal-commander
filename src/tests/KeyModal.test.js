import React from 'react';
import KeyModal from '../KeyModal';
import { shallow } from 'enzyme';

let wrapper;
describe('KeyModal', () => {

  beforeEach(() => {
    wrapper = shallow(
        <KeyModal toggleModal = { () => {} }/>
    );
  })

  it('should exist', () => {  
    expect(wrapper).toBeDefined()
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})