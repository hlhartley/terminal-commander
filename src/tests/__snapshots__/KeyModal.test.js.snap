// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyModal should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <KeyModal
    toggleModal={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="button-container"
        >
          <button
            onClick={[Function]}
          >
            x
          </button>
        </div>,
        <h2
          className="modal-title"
        >
          Key:
        </h2>,
        <ul
          className="modal-list"
        >
          <li>
            <span
              className="skyblue"
            >
              [file]
            </span>
             = file name
          </li>
          <li>
            <span
              className="skyblue"
            >
              [directory]
            </span>
             = directory name
          </li>
          <li>
            <span
              className="skyblue"
            >
              [file extension]
            </span>
             = file extension type (ex. .html, .css, .txt)
          </li>
        </ul>,
      ],
      "className": "key-modal",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            onClick={[Function]}
          >
            x
          </button>,
          "className": "button-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "x",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "x",
          "type": "button",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Key:",
          "className": "modal-title",
        },
        "ref": null,
        "rendered": "Key:",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <span
                className="skyblue"
              >
                [file]
              </span>
               = file name
            </li>,
            <li>
              <span
                className="skyblue"
              >
                [directory]
              </span>
               = directory name
            </li>,
            <li>
              <span
                className="skyblue"
              >
                [file extension]
              </span>
               = file extension type (ex. .html, .css, .txt)
            </li>,
          ],
          "className": "modal-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="skyblue"
                >
                  [file]
                </span>,
                " = file name",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "[file]",
                  "className": "skyblue",
                },
                "ref": null,
                "rendered": "[file]",
                "type": "span",
              },
              " = file name",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="skyblue"
                >
                  [directory]
                </span>,
                " = directory name",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "[directory]",
                  "className": "skyblue",
                },
                "ref": null,
                "rendered": "[directory]",
                "type": "span",
              },
              " = directory name",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="skyblue"
                >
                  [file extension]
                </span>,
                " = file extension type (ex. .html, .css, .txt)",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "[file extension]",
                  "className": "skyblue",
                },
                "ref": null,
                "rendered": "[file extension]",
                "type": "span",
              },
              " = file extension type (ex. .html, .css, .txt)",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="button-container"
          >
            <button
              onClick={[Function]}
            >
              x
            </button>
          </div>,
          <h2
            className="modal-title"
          >
            Key:
          </h2>,
          <ul
            className="modal-list"
          >
            <li>
              <span
                className="skyblue"
              >
                [file]
              </span>
               = file name
            </li>
            <li>
              <span
                className="skyblue"
              >
                [directory]
              </span>
               = directory name
            </li>
            <li>
              <span
                className="skyblue"
              >
                [file extension]
              </span>
               = file extension type (ex. .html, .css, .txt)
            </li>
          </ul>,
        ],
        "className": "key-modal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[Function]}
            >
              x
            </button>,
            "className": "button-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "x",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "x",
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Key:",
            "className": "modal-title",
          },
          "ref": null,
          "rendered": "Key:",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <span
                  className="skyblue"
                >
                  [file]
                </span>
                 = file name
              </li>,
              <li>
                <span
                  className="skyblue"
                >
                  [directory]
                </span>
                 = directory name
              </li>,
              <li>
                <span
                  className="skyblue"
                >
                  [file extension]
                </span>
                 = file extension type (ex. .html, .css, .txt)
              </li>,
            ],
            "className": "modal-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="skyblue"
                  >
                    [file]
                  </span>,
                  " = file name",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "[file]",
                    "className": "skyblue",
                  },
                  "ref": null,
                  "rendered": "[file]",
                  "type": "span",
                },
                " = file name",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="skyblue"
                  >
                    [directory]
                  </span>,
                  " = directory name",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "[directory]",
                    "className": "skyblue",
                  },
                  "ref": null,
                  "rendered": "[directory]",
                  "type": "span",
                },
                " = directory name",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="skyblue"
                  >
                    [file extension]
                  </span>,
                  " = file extension type (ex. .html, .css, .txt)",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "[file extension]",
                    "className": "skyblue",
                  },
                  "ref": null,
                  "rendered": "[file extension]",
                  "type": "span",
                },
                " = file extension type (ex. .html, .css, .txt)",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
